src/Server/HTTPHandler.cpp

```diff
@@ -588,11 +588,12 @@ void HTTPHandler::processQuery(
 
     /// At least, we should postpone sending of first buffer_size result bytes
     size_t buffer_size_total = std::max(
-        params.getParsed<size_t>("buffer_size", DBMS_DEFAULT_BUFFER_SIZE), static_cast<size_t>(DBMS_DEFAULT_BUFFER_SIZE));
+        std::max<size_t>(params.getParsed<size_t>("buffer_size", 0), context->getSettingsRef().http_response_buffer_size),
+        static_cast<size_t>(DBMS_DEFAULT_BUFFER_SIZE));
 
     /// If it is specified, the whole result will be buffered.
     ///  First ~buffer_size bytes will be buffered in memory, the remaining bytes will be stored in temporary file.
-    bool buffer_until_eof = params.getParsed<bool>("wait_end_of_query", false);
+    bool buffer_until_eof = params.getParsed<bool>("wait_end_of_query", false) || context->getSettingsRef().http_wait_end_of_query;
```

Comment:
This seems wrong. If `wait_end_of_query` is passed as an HTTP param I'd expect to superseed the server settings. Same for `buffer_size` above.

I guess the problem here is that in both cases you get a response whether they have been declared or not as part of the request (for both `wait_end_of_query` and `http_wait_end_of_query`).